### PR TITLE
feat(sdk): add kraken version specification

### PR DESCRIPTION
**Pull request by kuangxiang20240501**
## Description

- add kraken version specification
  - kraken version
  - product  compatible version
  - supported api use cases 

## Related Issue
https://github.com/mycloudnexus/kraken/issues/91

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)
- [ ] CI/CD or documentation update (changes to CI/CD pipeline or documentation)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

### CODE REVIEW

**Review by xuelianhan007 — DISMISSED** [review]
GOOD

File: kraken-java-sdk/kraken-java-sdk-controller/src/main/java/com/consoleconnect/kraken/operator/controller/api/KrakenApiUseCaseBuilderController.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  private String buildVersion;
+
+  private final SystemInfoService systemInfoService;
+  private final ApiComponentService apiComponentService;
+  private final UnifiedAssetRepository unifiedAssetRepository;
+
+  @SneakyThrows
+  @Operation(summary = "generate version specification")
+  @GetMapping("/version-specification")
+  public ResponseEntity<Mono<Resource>> versionSpecification() {
```
Comment (zhiyaoPCCW) [inline kraken-java-sdk/kraken-java-sdk-controller/src/main/java/com/consoleconnect/kraken/operator/controller/api/KrakenApiUseCaseBuilderController.java:61]: better to put complex logic into service layer.

**Review by xuelianhan007 — APPROVED** [review]
GOOD

**Review by xuelianhan007 — APPROVED** [review]
GOOD

**Review by zhiyaoPCCW — APPROVED** [review]
LGTM

**Pull request merged**